### ISSUE TITLE
Local Pickup options show in Cart block when shipping calculator is disabled

### ISSUE THREAD

**Issue by alexflorisca** [woothemes/woocommerce#52335]
> [!IMPORTANT]
> This issue is currently blocked by #50554, see https://github.com/woocommerce/woocommerce/issues/52335#issuecomment-2458614791.

### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the cart block, when the shipping calculator is disabled, the shipping options can't be changed. However local pickup doesn't need an address, so local pickup options show here. This is confusing for the shopper, as they can only change their shipping method to a local pickup one, but not a shipping one.

Slack conversation for context (p1729685919085089-slack-C02TS23QJ1X)

### Expected behavior

Ideally, we want to disable local pickup options from showing in the cart if the shipping calculator is disabled. 

### Actual behavior

Only the local pickup options show, but no shipping options.

### Steps to reproduce

1. In the admin navigate to WooCommerce > Settings > Shipping > Shipping settings and untick the `Enable the shipping calculator on the cart page` checkbox.
2. Make sure you have at least 1 pickup location enabled
3. On the frontend, add an item to cart
4. Visit the page with the cart block
5. Notice that only the pickup location is selectable on the cart page. 

### WordPress Environment

WC 9.4
WP 6.7 RC1

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by samueljseay** [comment]
Just mentioning, if it's possible, could we start this after: https://github.com/woocommerce/woocommerce/issues/50554

I have already run into an issue with another PR heavily refactoring this area that I will have to rebase into mine.
